### ISSUE TITLE
How to model a 'Procedure' (Annex II, SDG Regulation)

### ISSUE THREAD

**Issue by kareljanr** [catalogue-of-services-isa/CPSV-AP#42]
The Single Digital Gateway Regulation uses the term 'procedure' 184 times and contains the Annex II, which contains 21 procedures that should be fully digitalised by 2023. As this is a highly used concept, we propose to map this in the SDG common model. We have distinguished multiple approaches to do this:

1. A procedure is the same as a public service. In the CPSV-AP a public service is defined as: _A Public Service is a mandatory or discretionary set of activities performed, or able to be performed, by or on behalf of a public organisation, publicly funded and arise from public policy_ [...]

2. A procedure is a class, that is implemented by one or more public service(s).
![image](https://user-images.githubusercontent.com/47448176/67947561-31e95000-fbe4-11e9-9cec-1b882cbddb5c.png)

The Catalogue of Services project team's view: 
We prefer the second option for these reasons:

- The concept 'procedure' is differently named from a public service, which might cause confusion.
- Having both concepts separated will make it more clear and structured to map to procedures.
- It will also provide a more clear and structured framework for the future, when new procedures might be added.

**Comment by pwin** [comment]
... and a Procedure should perhaps be able to be linked to 0:* Procedure(s)

**Comment by kareljanr** [comment]
Japan is working on a similar solution (Universal Menu). In this solution they make a distinction between a procedure and a public service:

![image](https://user-images.githubusercontent.com/47448176/68292730-c2b0a780-008c-11ea-901c-ff7821302e22.png)

**Comment by anarosa-es** [comment]
Dear Kareljan,

Following the "`procedure`" definition (https://dictionary.cambridge.org/dictionary/english/procedure), it is a _**set of actions that is the official or accepted way of doing something**_, so it is composed by steps; on the other hand, "`service`" (https://dictionary.cambridge.org/dictionary/english/service) is always related to **_something that is done for people_**. In this sense, there are some official ways of doing something for people, i.e., "giving a birth certificate" can be officially done in several ways, at least, one way per Member State. The use of the word "procedure" in the Annex II is only a way to express that Member States should find the national procedures that match that **abstract** one. 

Therefore, it seems more correct to call the abstraction by "public service" and the "concretization" by "procedure". However, this doesn't fit with the CPSV-AP definition of "public service". So I propose to use the term "`SDG procedure`" for **the ones in Annex II** and "`procedure`" for the **national implementations of each "SDG procedure"**.

One more thing, in the reality of Public Administration, an "abstract procedure" without "concret procedures" has no sense.

**Comment by AGBielowski** [comment]
The definition offered by the SDG Regulation is not in line with the chosen perspective of both approaches.

Article 3(3):
‘procedure’ means a sequence of actions that must be taken by users to satisfy the requirements, or to obtain from a competent authority a decision, in order to be able to exercise their rights as referred to in point (a) of Article 2(2);

I am not really happy with the term "procedure" as it predictably leads to the present discussion, but I am satisfied that the Regulation adopts a user-centric perspective here: the actions taken by USERS. The SDG procedure term is consequently more related to terms like "user journey" than to terms like "service" or "process".

The perspective of the CPSV and the two approaches mentioned above by Karel is service provider-centric. Reconciling these two perspectives is a great opportunity offered by integrating SDG in CPSV. Probably we should use the new class "user procedure" to foreclose future discussion de nomine causa.

In a nutshell and in extension of approach (2), we should introduce a separate class, make the name specific to capture the user perspective, allow for the cardinalities 0..n and rename the relationship to "requires" or "is mapped to".

This opens two further questions:
1) At what level do we introduce country specificity? The SDG definition puts already the user procedure in the country specific domain (the actions and sequence thereof differs between member states) 
2) How do we give the notion of business-event and life-event a place in the core vocabulary? Is this the class that integrates between European definition and country specificity?
